plugins/base/src/main/kotlin/renderers/html/htmlPreprocessors.kt

```diff
@@ -63,6 +87,11 @@ object NavigationPageInstaller : PageTransformer {
                 children.filter { it is ContentPage && it.documentable is DEnumEntry }.map { visit(it as ContentPage) }
             else -> emptyList()
         }.sortedBy { it.name.toLowerCase() }
+
+    private tailrec fun flattenNavigationNodes(nodes: List<NavigationNode>, acc: List<NavigationNode> = emptyList()): List<NavigationNode> {
+        if(nodes.isEmpty()) return acc
+        return flattenNavigationNodes(nodes.flatMap { it.children }, nodes.map { it.copy(children = emptyList()) } + acc)
+    }
```

Comment:
This recursive function is really fancy but making `NavigationNode` implement `WithChildren` and then calling `.withDescendants` is simpler, more consistent and is guaranteed to be lazily evaluated.